### ISSUE TITLE
Coerce from right-to-left

### ISSUE THREAD

**Issue by marceloavf** [npm/node-semver#248]
By default, coerce runs from left-to-right returning the version, I would discuss if it should be possible to have an option to check from right-to-left.

Actual example: 
```js
semver.coerce('test/v1.0.0/app-v1.2.3-install.exe')
//=> version: "1.0.0"
```

Possibility
```js
semver.coerce('test/v1.0.0/app-v1.2.3-install.exe', 'right-to-left')
//=> version: "1.2.3"
```

**Comment by isaacs** [comment]
I'd avoid a magic string in the API, and we already accept an options object as the second argument.  How about `coerce(string, { rtl: true })`?

**Comment by marceloavf** [comment]
Would be nice @isaacs

**Comment by isaacs** [comment]
Landed on 6.2.0.  Turned out to be an interesting little algorithm!  You don't want `1.2.3.4` to coerce to `4.0.0` in rtl mode.  I'd expect `2.3.4` in that case.  I think what I landed on is reasonably efficient and does the right thing.
